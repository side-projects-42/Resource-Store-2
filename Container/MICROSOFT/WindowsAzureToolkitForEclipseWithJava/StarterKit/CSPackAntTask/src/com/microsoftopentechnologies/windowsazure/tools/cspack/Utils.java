/**
 * Copyright (c) Microsoft Corporation
 * 
 * All rights reserved. 
 * 
 * MIT License
 * 
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files 
 * (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, 
 * publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, 
 * subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF 
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR 
 * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH 
 * THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */


package com.microsoftopentechnologies.windowsazure.tools.cspack;

import com.github.jknack.handlebars.*;
import com.github.jknack.handlebars.context.JavaBeanValueResolver;
import com.microsoftopentechnologies.windowsazure.tools.build.WindowsAzurePackage;
import org.apache.tools.ant.BuildException;
import org.apache.tools.ant.util.Base64Converter;

import javax.xml.bind.JAXBContext;
import javax.xml.bind.JAXBException;
import javax.xml.bind.Unmarshaller;
import java.io.*;
import java.net.JarURLConnection;
import java.net.URL;
import java.net.URLConnection;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.*;
import java.util.jar.JarEntry;
import java.util.jar.JarFile;
import java.util.zip.ZipEntry;
import java.util.zip.ZipOutputStream;

public class Utils {

    private static String convertByteArrayToHexString(byte[] arrayBytes) {
        StringBuffer stringBuffer = new StringBuffer();
        for (int i = 0; i < arrayBytes.length; i++) {
            stringBuffer.append(Integer.toString((arrayBytes[i] & 0xff) + 0x100, 16)
                    .substring(1));
        }
        return stringBuffer.toString();
    }

    /**
     * Calculate Sha256 hash
     */
    public static String calcHash(File file) throws IOException {
        int bufferSize = 16384;
        String output;
        try {
            BufferedInputStream is = new BufferedInputStream(new FileInputStream(file));
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte [] buffer = new byte[bufferSize];
            int sizeRead = -1;
            while ((sizeRead = is.read(buffer)) != -1) {
                digest.update(buffer, 0, sizeRead);
            }
            is.close();

            byte [] hash = null;
            hash = new byte[digest.getDigestLength()];
            hash = digest.digest();
            output = new Base64Converter().encode(hash);
        } catch (FileNotFoundException e) {
            throw new BuildException(e);
        } catch (NoSuchAlgorithmException e) {
            throw new BuildException(e);
        }
        return output;
    }

    public static String calcHexHash(File file) throws IOException {
        int bufferSize = 16384;
        String output;
        try {
            BufferedInputStream is = new BufferedInputStream(new FileInputStream(file));
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte [] buffer = new byte[bufferSize];
            int sizeRead = -1;
            while ((sizeRead = is.read(buffer)) != -1) {
                digest.update(buffer, 0, sizeRead);
            }
            is.close();

            byte [] hash = null;
            hash = new byte[digest.getDigestLength()];
            hash = digest.digest();
            output = convertByteArrayToHexString(hash).toUpperCase();
        } catch (FileNotFoundException e) {
            throw new BuildException(e);
        } catch (NoSuchAlgorithmException e) {
            throw new BuildException(e);
        }
        return output;
    }

    /**
     * Zip directory (source can't be a file)
     */
    public static void zipDirectory(File sourceDir, File zipFile, WindowsAzurePackage waPackage) {
        for (File file : sourceDir.listFiles()) {
            waPackage.zipFile(file, zipFile);
            delete(file);
        }
    }

    /**
     * Compresses the given directory and all its sub-directories into a ZIP file.
     * <p>
     * The ZIP file must not be a directory and its parent directory must exist.
     *
     * @param sourceDir
     *          root directory.
     * @param targetZipName
     *          name of ZIP file that will be created or overwritten.
     */
    public static void pack(File sourceDir, String targetZipName, boolean deleteSource) {
        if (!sourceDir.exists()) {
            throw new RuntimeException("Given file '" + sourceDir + "' doesn't exist!");
        }
        ZipOutputStream out = null;
        try {
            String[] filenames = sourceDir.list();
            if (filenames == null) {
                if (!sourceDir.exists()) {
                    throw new RuntimeException("Given file '" + sourceDir + "' doesn't exist!");
                }
                throw new IOException("Given file is not a directory '" + sourceDir + "'");
            }
            out = new ZipOutputStream(new BufferedOutputStream(new FileOutputStream(new File(targetZipName))));
            pack(filenames, sourceDir, out, "");
            if (deleteSource) {
                for (String filename : filenames) {
                    delete(new File(sourceDir, filename));
                }
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        } finally {
            closeQuietly(out);
        }
    }

    /**
     * Compresses the given directory and all its sub-directories into a ZIP file.
     *
     * @param filenames
     *          files and directories names.
     * @param dir
     *          root directory.
     * @param out
     *          ZIP output stream.
     * @param pathPrefix
     *          prefix to be used for the entries.
     */
    private static void pack(String[] filenames, File dir, ZipOutputStream out, String pathPrefix) throws IOException {
        for (int i = 0; i < filenames.length; i++) {
            String filename = filenames[i];
            File file = new File(dir, filename);
            boolean isDir = file.isDirectory();
            String path = pathPrefix + file.getName();
            if (isDir) {
                path += "/";
            }
            // Create a ZIP entry
            ZipEntry zipEntry = new ZipEntry(path);
            if (!isDir) {
                zipEntry.setSize(file.length());
                zipEntry.setTime(file.lastModified());
            }
            out.putNextEntry(zipEntry);
            // Copy the file content
            if (!isDir) {
                copy(file, out);
            }
            out.closeEntry();
            // Traverse the directory
            if (isDir) {
                String[] dirFilenames = file.list();
                if (dirFilenames == null) {
                    if (!dir.exists()) {
                        throw new RuntimeException("Given file '" + dir + "' doesn't exist!");
                    }
                    throw new IOException("Given file is not a directory '" + dir + "'");
                }
                pack(dirFilenames, file, out, path);
            }
        }
    }

    /**
     * Copies the given file into an output stream.
     *
     * @param file input file (must exist).
     * @param out output stream.
     */
    public static void copy(File file, OutputStream out) throws IOException {
        FileInputStream in = new FileInputStream(file);
        try {
            byte[] buffer = new byte[4*1024];
            long count = 0;
            int n = 0;
            InputStream input = new BufferedInputStream(in);
            while (-1 != (n = input.read(buffer))) {
                out.write(buffer, 0, n);
                count += n;
            }
        }
        finally {
            closeQuietly(in);
        }
    }

    /**
     * Unconditionally close an <code>InputStream</code>.
     * @param stream the InputStream to close, may be null or already closed
     */
    public static void closeQuietly(Closeable stream) {
        try {
            if (stream != null) {
                stream.close();
            }
        } catch (IOException ioe) {
            // ignore
        }
    }

    public static void delete(File f) {
        if (f.isDirectory()) {
            for (File c : f.listFiles())
                delete(c);
        }
        if (!f.delete())
            throw new RuntimeException("Failed to delete file: " + f);
    }

    public static void generateFileList(Collection<File> fileList, File node) {
        //add file only
        if (node.isFile()) {
            fileList.add(node.getAbsoluteFile());
        }

        if (node.isDirectory()) {
            String[] subNote = node.list();
            for (String filename : subNote) {
                generateFileList(fileList, new File(node, filename));
            }
        }
    }

    public static String generateUID(String name) {
        UUID uuid = UUID.nameUUIDFromBytes(name.getBytes());
        return uuid.toString().replaceAll("-", "");
    }

    static void createDirectory(String name) {
        File serviceDefinitionDirectory = new File(name);
        if (!serviceDefinitionDirectory.exists()) {
            serviceDefinitionDirectory.mkdir();
        }
    }

    public static void applyTemplateWithPath(String templateName, Object model, String deployPath) throws IOException {
        applyTemplate(templateName, model, deployPath + File.separator + templateName);
    }

    public static void applyTemplate(String templateName, Object model, String fileName) throws IOException {
//        System.out.println("templateName=" + templateName + "; fileName=" + fileName);
        BufferedWriter fileWriter = new BufferedWriter(new FileWriter(fileName));
        try {
            Handlebars handlebars = new Handlebars();
            Template template = handlebars.compile("/templates/" + templateName);

            Context context = Context.newBuilder(model).resolver(JavaBeanValueResolver.INSTANCE).build();
            template.apply(context, fileWriter);
//            System.out.println(template.apply(context));
        } finally {
            fileWriter.flush();
            fileWriter.close();
        }
    }

    public static List<String> getJarEntries(String jarName, String entryName) throws IOException {
        List<String> files = new ArrayList<String>();
        System.out.println("entryName = " + entryName);
        JarURLConnection urlConnection = (JarURLConnection) new URL("jar:file:" + jarName + "!/" + entryName).openConnection();
//        URLConnection urlConnection = originUrl.openConnection();
            JarURLConnection jarConnection = ((JarURLConnection)urlConnection);
            JarFile jarFile = jarConnection.getJarFile();
            Enumeration<JarEntry> entries = jarFile.entries();

            while (entries.hasMoreElements()) {
                JarEntry entry = entries.nextElement();
                if (entry.getName().startsWith(jarConnection.getEntryName())) {
                    if (!entry.isDirectory()) {
                        files.add(entry.getName());
                    }
                }
            }
        return files;
    }

    public static void copyJarEntry(String source, File destination) throws IOException {
        InputStream stream = Utils.class.getResourceAsStream(source);
        if (stream == null) {
            throw new BuildException(source + " not found");
        }
        OutputStream resStreamOut = null;
        int readBytes;
        byte[] buffer = new byte[4096];
        try {
            resStreamOut = new FileOutputStream(destination);
            while ((readBytes = stream.read(buffer)) > 0) {
                resStreamOut.write(buffer, 0, readBytes);
            }
        } finally {
            stream.close();
            if (resStreamOut != null) {
                resStreamOut.close();
            }
        }
    }

    public static <T> T parseXmlFile(Class<T> type, String filePath) throws JAXBException {
        T instance = null;
        File file = new File(filePath);
        JAXBContext jaxbContext = JAXBContext.newInstance(type);

        Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
        instance = (T) jaxbUnmarshaller.unmarshal(file);
        return instance;
    }
}
