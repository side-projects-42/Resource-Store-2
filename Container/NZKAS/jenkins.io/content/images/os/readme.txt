Files in this directory are taken from VirtualBox at http://www.virtualbox.org/svn/vbox/trunk/src/VBox/Frontends/VirtualBox/images

Copyrighted by Sun Microsystems under GPL. See COPYING.
