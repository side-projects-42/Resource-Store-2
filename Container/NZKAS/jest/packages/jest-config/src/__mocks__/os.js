/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

'use strict';

const os = jest.genMockFromModule('os');

let cpus;
function __setCpus(newCpus) {
  cpus = newCpus;
}

os.__setCpus = __setCpus;
os.cpus = jest.fn(() => cpus);

module.exports = os;
