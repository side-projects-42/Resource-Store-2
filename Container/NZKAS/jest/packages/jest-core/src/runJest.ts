/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import chalk from 'chalk';
import {sync as realpath} from 'realpath-native';
import {CustomConsole} from '@jest/console';
import {formatTestResults} from 'jest-util';
import exit from 'exit';
import fs from 'graceful-fs';
import {JestHook, JestHookEmitter} from 'jest-watcher';
import {Context} from 'jest-runtime';
import {Test} from 'jest-runner';
import {Config} from '@jest/types';
import {
  AggregatedResult,
  makeEmptyAggregatedTestResult,
} from '@jest/test-result';
import {ChangedFiles, ChangedFilesPromise} from 'jest-changed-files';
import getNoTestsFoundMessage from './getNoTestsFoundMessage';
import runGlobalHook from './runGlobalHook';
import SearchSource from './SearchSource';
import TestScheduler, {TestSchedulerContext} from './TestScheduler';
import TestSequencer from './TestSequencer';
import FailedTestsCache from './FailedTestsCache';
import collectNodeHandles from './collectHandles';
import TestWatcher from './TestWatcher';
import {TestRunData, Filter} from './types';

const getTestPaths = async (
  globalConfig: Config.GlobalConfig,
  context: Context,
  outputStream: NodeJS.WritableStream,
  changedFiles: ChangedFiles | undefined,
  jestHooks: JestHookEmitter,
  filter?: Filter,
) => {
  const source = new SearchSource(context);
  const data = await source.getTestPaths(globalConfig, changedFiles, filter);

  if (!data.tests.length && globalConfig.onlyChanged && data.noSCM) {
    new CustomConsole(outputStream, outputStream).log(
      'Jest can only find uncommitted changed files in a git or hg ' +
        'repository. If you make your project a git or hg ' +
        'repository (`git init` or `hg init`), Jest will be able ' +
        'to only run tests related to files changed since the last ' +
        'commit.',
    );
  }

  const shouldTestArray = await Promise.all(
    data.tests.map(test =>
      jestHooks.shouldRunTestSuite({
        config: test.context.config,
        duration: test.duration,
        testPath: test.path,
      }),
    ),
  );

  const filteredTests = data.tests.filter((_test, i) => shouldTestArray[i]);

  return {...data, allTests: filteredTests.length, tests: filteredTests};
};

type ProcessResultOptions = Pick<
  Config.GlobalConfig,
  'json' | 'outputFile' | 'testResultsProcessor'
> & {
  collectHandles?: () => Array<Error>;
  onComplete?: (result: AggregatedResult) => void;
  outputStream: NodeJS.WritableStream;
};

const processResults = (
  runResults: AggregatedResult,
  options: ProcessResultOptions,
) => {
  const {
    outputFile,
    json: isJSON,
    onComplete,
    outputStream,
    testResultsProcessor,
    collectHandles,
  } = options;

  if (collectHandles) {
    runResults.openHandles = collectHandles();
  } else {
    runResults.openHandles = [];
  }

  if (testResultsProcessor) {
    runResults = require(testResultsProcessor)(runResults);
  }
  if (isJSON) {
    if (outputFile) {
      const cwd = realpath(process.cwd());
      const filePath = path.resolve(cwd, outputFile);

      fs.writeFileSync(filePath, JSON.stringify(formatTestResults(runResults)));
      outputStream.write(
        `Test results written to: ${path.relative(cwd, filePath)}\n`,
      );
    } else {
      process.stdout.write(JSON.stringify(formatTestResults(runResults)));
    }
  }

  return onComplete && onComplete(runResults);
};

const testSchedulerContext: TestSchedulerContext = {
  firstRun: true,
  previousSuccess: true,
};

export default (async function runJest({
  contexts,
  globalConfig,
  outputStream,
  testWatcher,
  jestHooks = new JestHook().getEmitter(),
  startRun,
  changedFilesPromise,
  onComplete,
  failedTestsCache,
  filter,
}: {
  globalConfig: Config.GlobalConfig;
  contexts: Array<Context>;
  outputStream: NodeJS.WritableStream;
  testWatcher: TestWatcher;
  jestHooks?: JestHookEmitter;
  startRun: (globalConfig: Config.GlobalConfig) => void;
  changedFilesPromise?: ChangedFilesPromise;
  onComplete: (testResults: AggregatedResult) => void;
  failedTestsCache?: FailedTestsCache;
  filter?: Filter;
}) {
  const sequencer = new TestSequencer();
  let allTests: Array<Test> = [];

  if (changedFilesPromise && globalConfig.watch) {
    const {repos} = await changedFilesPromise;

    const noSCM = (Object.keys(repos) as Array<
      keyof ChangedFiles['repos']
    >).every(scm => repos[scm].size === 0);
    if (noSCM) {
      process.stderr.write(
        '\n' +
          chalk.bold('--watch') +
          ' is not supported without git/hg, please use --watchAll ' +
          '\n',
      );
      exit(1);
    }
  }

  const testRunData: TestRunData = await Promise.all(
    contexts.map(async context => {
      const matches = await getTestPaths(
        globalConfig,
        context,
        outputStream,
        changedFilesPromise && (await changedFilesPromise),
        jestHooks,
        filter,
      );
      allTests = allTests.concat(matches.tests);

      return {context, matches};
    }),
  );

  allTests = sequencer.sort(allTests);

  if (globalConfig.listTests) {
    const testsPaths = Array.from(new Set(allTests.map(test => test.path)));
    if (globalConfig.json) {
      console.log(JSON.stringify(testsPaths));
    } else {
      console.log(testsPaths.join('\n'));
    }

    onComplete && onComplete(makeEmptyAggregatedTestResult());
    return null;
  }

  if (globalConfig.onlyFailures && failedTestsCache) {
    allTests = failedTestsCache.filterTests(allTests);
    globalConfig = failedTestsCache.updateConfig(globalConfig);
  }

  const hasTests = allTests.length > 0;

  if (!hasTests) {
    const noTestsFoundMessage = getNoTestsFoundMessage(
      testRunData,
      globalConfig,
    );

    if (
      globalConfig.passWithNoTests ||
      globalConfig.findRelatedTests ||
      globalConfig.lastCommit ||
      globalConfig.onlyChanged
    ) {
      new CustomConsole(outputStream, outputStream).log(noTestsFoundMessage);
    } else {
      new CustomConsole(outputStream, outputStream).error(noTestsFoundMessage);

      exit(1);
    }
  } else if (
    allTests.length === 1 &&
    globalConfig.silent !== true &&
    globalConfig.verbose !== false
  ) {
    const newConfig: Config.GlobalConfig = {...globalConfig, verbose: true};
    globalConfig = Object.freeze(newConfig);
  }

  let collectHandles;

  if (globalConfig.detectOpenHandles) {
    collectHandles = collectNodeHandles();
  }

  if (hasTests) {
    await runGlobalHook({allTests, globalConfig, moduleName: 'globalSetup'});
  }

  if (changedFilesPromise) {
    testSchedulerContext.changedFiles = (await changedFilesPromise).changedFiles;
  }

  const results = await new TestScheduler(
    globalConfig,
    {startRun},
    testSchedulerContext,
  ).scheduleTests(allTests, testWatcher);

  sequencer.cacheResults(allTests, results);

  if (hasTests) {
    await runGlobalHook({allTests, globalConfig, moduleName: 'globalTeardown'});
  }

  return processResults(results, {
    collectHandles,
    json: globalConfig.json,
    onComplete,
    outputFile: globalConfig.outputFile,
    outputStream,
    testResultsProcessor: globalConfig.testResultsProcessor,
  });
});
