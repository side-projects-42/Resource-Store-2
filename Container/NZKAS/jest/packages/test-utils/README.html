<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
  <head>
    <meta charset="utf-8" />
    <meta name="generator" content="pandoc" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1.0, user-scalable=yes"
    />
    <title>README</title>
    <style type="text/css">
      code {
        white-space: pre-wrap;
      }
      span.smallcaps {
        font-variant: small-caps;
      }
      span.underline {
        text-decoration: underline;
      }
      div.column {
        display: inline-block;
        vertical-align: top;
        width: 50%;
      }
    </style>
    <style type="text/css">
      a.sourceLine {
        display: inline-block;
        line-height: 1.25;
      }
      a.sourceLine {
        pointer-events: none;
        color: inherit;
        text-decoration: inherit;
      }
      a.sourceLine:empty {
        height: 1.2em;
      }
      .sourceCode {
        overflow: visible;
      }
      code.sourceCode {
        white-space: pre;
        position: relative;
      }
      div.sourceCode {
        margin: 1em 0;
      }
      pre.sourceCode {
        margin: 0;
      }
      @media screen {
        div.sourceCode {
          overflow: auto;
        }
      }
      @media print {
        code.sourceCode {
          white-space: pre-wrap;
        }
        a.sourceLine {
          text-indent: -1em;
          padding-left: 1em;
        }
      }
      pre.numberSource a.sourceLine {
        position: relative;
        left: -4em;
      }
      pre.numberSource a.sourceLine::before {
        content: attr(title);
        position: relative;
        left: -1em;
        text-align: right;
        vertical-align: baseline;
        border: none;
        pointer-events: all;
        display: inline-block;
        -webkit-touch-callout: none;
        -webkit-user-select: none;
        -khtml-user-select: none;
        -moz-user-select: none;
        -ms-user-select: none;
        user-select: none;
        padding: 0 4px;
        width: 4em;
        color: #aaaaaa;
      }
      pre.numberSource {
        margin-left: 3em;
        border-left: 1px solid #aaaaaa;
        padding-left: 4px;
      }
      div.sourceCode {
      }
      @media screen {
        a.sourceLine::before {
          text-decoration: underline;
        }
      }
      code span.al {
        color: #ff0000;
        font-weight: bold;
      } /* Alert */
      code span.an {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Annotation */
      code span.at {
        color: #7d9029;
      } /* Attribute */
      code span.bn {
        color: #40a070;
      } /* BaseN */
      code span.bu {
      } /* BuiltIn */
      code span.cf {
        color: #007020;
        font-weight: bold;
      } /* ControlFlow */
      code span.ch {
        color: #4070a0;
      } /* Char */
      code span.cn {
        color: #880000;
      } /* Constant */
      code span.co {
        color: #60a0b0;
        font-style: italic;
      } /* Comment */
      code span.cv {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* CommentVar */
      code span.do {
        color: #ba2121;
        font-style: italic;
      } /* Documentation */
      code span.dt {
        color: #902000;
      } /* DataType */
      code span.dv {
        color: #40a070;
      } /* DecVal */
      code span.er {
        color: #ff0000;
        font-weight: bold;
      } /* Error */
      code span.ex {
      } /* Extension */
      code span.fl {
        color: #40a070;
      } /* Float */
      code span.fu {
        color: #06287e;
      } /* Function */
      code span.im {
      } /* Import */
      code span.in {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Information */
      code span.kw {
        color: #007020;
        font-weight: bold;
      } /* Keyword */
      code span.op {
        color: #666666;
      } /* Operator */
      code span.ot {
        color: #007020;
      } /* Other */
      code span.pp {
        color: #bc7a00;
      } /* Preprocessor */
      code span.sc {
        color: #4070a0;
      } /* SpecialChar */
      code span.ss {
        color: #bb6688;
      } /* SpecialString */
      code span.st {
        color: #4070a0;
      } /* String */
      code span.va {
        color: #19177c;
      } /* Variable */
      code span.vs {
        color: #4070a0;
      } /* VerbatimString */
      code span.wa {
        color: #60a0b0;
        font-weight: bold;
        font-style: italic;
      } /* Warning */
    </style>
  </head>
  <body>
    <h1 id="babel-jest">babel-jest</h1>
    <p>
      <a href="https://github.com/babel/babel">Babel</a>
      <a href="https://github.com/facebook/jest">jest</a> plugin
    </p>
    <h2 id="usage">Usage</h2>
    <p>
      If you are already using <code>jest-cli</code>, just add
      <code>babel-jest</code> and it will automatically compile JavaScript code
      using Babel.
    </p>
    <div class="sourceCode" id="cb1">
      <pre
        class="sourceCode bash"
      ><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" title="1"><span class="ex">yarn</span> add --dev babel-jest @babel/core</a></code></pre>
    </div>
    <p>
      If you would like to write your own preprocessor, uninstall and delete
      babel-jest and set the
      <a
        href="https://jestjs.io/docs/configuration#transform-object-string-string"
        >config.transform</a
      >
      option to your preprocessor.
    </p>
    <h2 id="setup">Setup</h2>
    <p>
      <em
        >Note: this step is only required if you are using
        <code>babel-jest</code> with additional code preprocessors.</em
      >
    </p>
    <p>
      To explicitly define <code>babel-jest</code> as a transformer for your
      JavaScript code, map <em>.js</em> files to the
      <code>babel-jest</code> module.
    </p>
    <div class="sourceCode" id="cb2">
      <pre
        class="sourceCode json"
      ><code class="sourceCode json"><a class="sourceLine" id="cb2-1" title="1"><span class="er">&quot;transform&quot;:</span> <span class="fu">{</span></a>
<a class="sourceLine" id="cb2-2" title="2">  <span class="dt">&quot;^.+</span><span class="ch">\\</span><span class="dt">.jsx?$&quot;</span><span class="fu">:</span> <span class="st">&quot;babel-jest&quot;</span></a>
<a class="sourceLine" id="cb2-3" title="3"><span class="fu">}</span><span class="er">,</span></a></code></pre>
    </div>
  </body>
</html>
