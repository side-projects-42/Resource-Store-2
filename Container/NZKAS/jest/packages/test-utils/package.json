{
  "name": "@jest/test-utils",
  "version": "24.4.0",
  "private": true,
  "license": "MIT",
  "main": "build/index.js",
  "types": "build/index.d.ts"
}
