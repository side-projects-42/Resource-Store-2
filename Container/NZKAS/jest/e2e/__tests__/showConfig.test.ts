/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import os from 'os';
import {wrap} from 'jest-snapshot-serializer-raw';
import {skipSuiteOnWindows} from '@jest/test-utils';
import runJest from '../runJest';
import {cleanup, writeFiles} from '../Utils';

skipSuiteOnWindows();

const DIR = path.resolve(os.tmpdir(), 'show-config-test');

beforeEach(() => cleanup(DIR));
afterEach(() => cleanup(DIR));

test('--showConfig outputs config info and exits', () => {
  writeFiles(DIR, {
    '__tests__/test.test.js': `test('test', () => {});`,
    'package.json': JSON.stringify({jest: {environment: 'node'}}),
  });

  let {stdout} = runJest(DIR, [
    '--showConfig',
    '--no-cache',
    // Make the snapshot flag stable on CI.
    '--updateSnapshot',
  ]);

  stdout = stdout
    .replace(/"cacheDirectory": "(.+)"/g, '"cacheDirectory": "/tmp/jest"')
    .replace(/"name": "(.+)"/g, '"name": "[md5 hash]"')
    .replace(/"version": "(.+)"/g, '"version": "[version]"')
    .replace(/"maxWorkers": (\d+)/g, '"maxWorkers": "[maxWorkers]"')
    .replace(/\"\S*show-config-test/gm, '"<<REPLACED_ROOT_DIR>>')
    .replace(/\"\S*\/jest\/packages/gm, '"<<REPLACED_JEST_PACKAGES_DIR>>');

  expect(wrap(stdout)).toMatchSnapshot();
});
