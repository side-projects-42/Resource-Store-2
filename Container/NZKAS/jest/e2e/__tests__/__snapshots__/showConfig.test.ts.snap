// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--showConfig outputs config info and exits 1`] = `
{
  "configs": [
    {
      "automock": false,
      "browser": false,
      "cache": false,
      "cacheDirectory": "/tmp/jest",
      "clearMocks": false,
      "coveragePathIgnorePatterns": [
        "/node_modules/"
      ],
      "cwd": "<<REPLACED_ROOT_DIR>>",
      "dependencyExtractor": null,
      "detectLeaks": false,
      "detectOpenHandles": false,
      "errorOnDeprecated": false,
      "filter": null,
      "forceCoverageMatch": [],
      "globalSetup": null,
      "globalTeardown": null,
      "globals": {},
      "haste": {
        "computeSha1": false,
        "providesModuleNodeModules": [],
        "throwOnModuleCollision": false
      },
      "moduleDirectories": [
        "node_modules"
      ],
      "moduleFileExtensions": [
        "js",
        "json",
        "jsx",
        "ts",
        "tsx",
        "node"
      ],
      "moduleNameMapper": {},
      "modulePathIgnorePatterns": [],
      "name": "[md5 hash]",
      "prettierPath": "prettier",
      "resetMocks": false,
      "resetModules": false,
      "resolver": null,
      "restoreMocks": false,
      "rootDir": "<<REPLACED_ROOT_DIR>>",
      "roots": [
        "<<REPLACED_ROOT_DIR>>"
      ],
      "runner": "jest-runner",
      "setupFiles": [],
      "setupFilesAfterEnv": [],
      "skipFilter": false,
      "snapshotSerializers": [],
      "testEnvironment": "<<REPLACED_JEST_PACKAGES_DIR>>/jest-environment-jsdom/build/index.js",
      "testEnvironmentOptions": {},
      "testLocationInResults": false,
      "testMatch": [
        "**/__tests__/**/*.[jt]s?(x)",
        "**/?(*.)+(spec|test).[tj]s?(x)"
      ],
      "testPathIgnorePatterns": [
        "/node_modules/"
      ],
      "testRegex": [],
      "testRunner": "<<REPLACED_JEST_PACKAGES_DIR>>/jest-jasmine2/build/index.js",
      "testURL": "http://localhost",
      "timers": "real",
      "transform": [
        [
          "^.+\\\\.[jt]sx?$",
          "<<REPLACED_JEST_PACKAGES_DIR>>/babel-jest/build/index.js"
        ]
      ],
      "transformIgnorePatterns": [
        "/node_modules/"
      ],
      "watchPathIgnorePatterns": []
    }
  ],
  "globalConfig": {
    "bail": 0,
    "changedFilesWithAncestor": false,
    "collectCoverage": false,
    "collectCoverageFrom": null,
    "coverageDirectory": "<<REPLACED_ROOT_DIR>>/coverage",
    "coverageReporters": [
      "json",
      "text",
      "lcov",
      "clover"
    ],
    "coverageThreshold": null,
    "detectLeaks": false,
    "detectOpenHandles": false,
    "errorOnDeprecated": false,
    "expand": false,
    "filter": null,
    "globalSetup": null,
    "globalTeardown": null,
    "json": false,
    "listTests": false,
    "maxConcurrency": 5,
    "maxWorkers": "[maxWorkers]",
    "noStackTrace": false,
    "nonFlagArgs": [],
    "notify": false,
    "notifyMode": "failure-change",
    "passWithNoTests": false,
    "projects": null,
    "rootDir": "<<REPLACED_ROOT_DIR>>",
    "runTestsByPath": false,
    "skipFilter": false,
    "testFailureExitCode": 1,
    "testPathPattern": "",
    "testResultsProcessor": null,
    "updateSnapshot": "all",
    "useStderr": false,
    "verbose": null,
    "watch": false,
    "watchman": true
  },
  "version": "[version]"
}
`;
